src/sdk/namespace_fs.js

```diff
@@ -747,9 +752,22 @@ class NamespaceFS {
                 }
                 // TODO use file xattr to store md5_b64 xattr, etc.
                 const stat = await nb_native().fs.stat(fs_account_config, upload_path);
+                const is_gpfs = Boolean(fs_account_config.backend === 'GPFS' && nb_native().fs.gpfs);
+                const is_encrypted = is_gpfs && await nb_native().fs.is_encrypted(fs_account_config, upload_path);
+                if (params.encryption && params.encryption.algorithm === 'AES256' && !is_encrypted) {
+                    dbg.error('upload_object: User requested encryption but encryption not supported for FS');
+                    throw new Error('upload_object: User requested encryption but encryption not supported for FS');
```

Comment:
This will become an internal error in S3, right. We want a more suitable error like ServerSideEncryptionConfigurationNotFoundError ([see](https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html#ErrorCodeList)) and also we want to add a custom message to explain the situation to the user better.